diagnostic.md

```diff
@@ -3,36 +3,47 @@
 What are three advantages of using Ember?
 
 ```sh
-# your answer here
+- Ember provides a complete solution when it comes to a client side app.
+- The app runs within the users browser and is self contained, including everything it needs.
+- Ember is meant to be completely backward compatible so that past, or even outdated, innovations
+  are maintained.
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
 ```sh
-# your answer here
+Ember uses Handlebars.  Handlebars is a templating engine that is also
+maintained by the Emberteam
+
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
 ```sh
-# your answer here
+- Ember-Data is ember’s storage layer.
+- The Ember’s Views layer manages the rendering of views that connect to the browsers DOM
 ```
 
 How does a component differ from an Ember template?
 
 ```sh
-# your answer here
+An Ember component consists of a Handlebars template file and an accompanying Ember class.
+The implementation of this class is required only when we need extra interactivity with
+the component. A component is used like ordinary HTML tag.
 ```
 
 What is an opinionated framework? Is Ember opinionated? How or how not?
 
 ```sh
-# your answer here
+An opinionated framework can be  limiting and encourages the designer to do things their way.
+Ember is considered a very opinionated framework which forces you to structure your app the way
+it wants you to.
 ```
 
 In Ember what is a route's job (please list three)?
 
 ```sh
-# your answer here
+Routes display the appropriate model depending on the hook. This is then handed over to the component
+for display
```

Comment:
Yes! The one other thing it does is parse data from the URL, such as an id.
.5/1